exporter/datadogexporter/internal/testutils/test_utils.go

```diff
@@ -39,18 +52,21 @@ var (
 type DatadogServer struct {
 	*httptest.Server
 	MetadataChan chan []byte
+	// LogsData is the array of json requests sent to datadog backend
+	LogsData []map[string]interface{}
 }
 
 // DatadogServerMock mocks a Datadog backend server
 func DatadogServerMock(overwriteHandlerFuncs ...OverwriteHandleFunc) *DatadogServer {
 	metadataChan := make(chan []byte)
 	mux := http.NewServeMux()
 
+	server := &DatadogServer{}
 	handlers := map[string]http.HandlerFunc{
 		"/api/v1/validate": validateAPIKeyEndpoint,
 		"/api/v1/series":   metricsEndpoint,
 		"/intake":          newMetadataEndpoint(metadataChan),
-		"/":                func(w http.ResponseWriter, r *http.Request) {},
+		"/":                server.logsEndPoint,
```

Comment:
We should use a different struct for this. `DatadogServer` mocks `api.datadoghq.com`, and here you are mocking `http-intake.datadoghq.com` in the same struct. This will lead to very confusing errors and worse test coverage if we ever send payloads to `/` on `api.datadoghq.com`.

Reply:
I have created different mock server for logs